src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1644,16 +1649,61 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
                 }
             }
 
-            // Finally, if one has fewer custom modifiers, that is better
+            // Otherwise, if one has fewer custom modifiers, that is better
             int m1ModifierCount = m1.LeastOverriddenMember.CustomModifierCount();
             int m2ModifierCount = m2.LeastOverriddenMember.CustomModifierCount();
             if (m1ModifierCount != m2ModifierCount)
             {
                 return (m1ModifierCount < m2ModifierCount) ? BetterResult.Left : BetterResult.Right;
             }
 
-            // Otherwise, neither function member is better.
-            return BetterResult.Neither;
+            // Otherwise, prefer methods with 'val' parameters over 'in' parameters.
+            return PreferValOverInParameters(arguments, m1, m1LeastOverridenParameters, m2, m2LeastOverridenParameters);
+        }
+
+        private static BetterResult PreferValOverInParameters<TMember>(
```

Comment:
>PreferValOverInParameters [](start = 36, length = 25)

Are all code paths in this function covered by tests?

Reply:
Yes.

---
In reply to: [155010398](https://github.com/dotnet/roslyn/pull/23122#discussion_r155010398) [](ancestors = 155010398)